DomainContent/AvatarStore/attachmentItemScript.js

```diff
@@ -0,0 +1,156 @@
+//
+//  attachmentItemScript.js
+//
+//  This script is a simplified version of the original attachmentItemScript.js 
+//  Copyright 2017 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+(function() {
+    var ATTACH_SOUND = SoundCache.getSound(Script.resolvePath('sound/attach_sound_1.wav'));
+    var DETACH_SOUND = SoundCache.getSound(Script.resolvePath('sound/detach.wav'));
+
+    var LEFT_RIGHT_PLACEHOLDER = '[LR]';
+    var ATTACH_DISTANCE = 0.35;
+    var DETACH_DISTANCE = 0.5;
+    var AUDIO_VOLUME_LEVEL = 0.2;
+
+    var TRIGGER_INTENSITY = 1.0;
+    var TRIGGER_TIME = 0.2;
+
+    var EMPTY_PARENT_ID = "{00000000-0000-0000-0000-000000000000}";
+
+    var MESSAGE_CHANNEL_BASE = "AvatarStoreObject";
+    var messageChannel;
+    
+    var _this, _entityID;
+    var _attachmentData;
+    var _supportedJoints = [];
+    var isAttached;
+
+    var firstGrab = true;
+
+    /**
+     * 
+     * @param {Object} entityProperties 
+     * @param {touchJSONUserDataCallback} touchCallback 
+     */
+    var touchJSONUserData = function(entityProperties, touchCallback) {
+        try {
+            // attempt to touch the userData
+            var userData = JSON.parse(entityProperties.userData);
+            touchCallback.call(this, userData);
+            entityProperties.userData = JSON.stringify(userData);
+        } catch (e) {
+            print('Something went wrong while trying to touch/modify the userData. Could be invalid JSON or problem with the callback function.');
+        }
+    };
+
+    /**
+     * This callback is displayed as a global member.
+     * @callback touchJSONUserDataCallback
+     * @param {Object} userData
+     */
+
+    function AttachableItem() {
+        _this = this;
+    }
+    AttachableItem.prototype = {
+        preload : function(entityID) {
+            _entityID = entityID;
+            var properties = Entities.getEntityProperties(entityID);
+
+            _attachmentData = JSON.parse(properties.userData).Attachment;
+            if (_attachmentData.joint.indexOf(LEFT_RIGHT_PLACEHOLDER) !== -1) {
+                var baseJoint = _attachmentData.joint.substring(4);
+                _supportedJoints.push("Left".concat(baseJoint));
+                _supportedJoints.push("Right".concat(baseJoint));
+            } else {
+                _supportedJoints.push(_attachmentData.joint);
+            }
+
+            isAttached = _attachmentData.attached;
+
+            if (Entities.getNestableType(properties.parentID) !== "avatar" && !isAttached) {
+                messageChannel = MESSAGE_CHANNEL_BASE + properties.parentID;
+                Messages.subscribe(messageChannel);
+            }
+        },
+        startNearGrab: function(entityID, args) {
+            if (firstGrab) {
+                var properties = Entities.getEntityProperties(entityID);  
```

Comment:
you could change this to `var isVisible = Entities.getEntityProperties(entityID, 'visible').visible;`